client/src/components/FeatureCalloutWidget.tsx

```diff
@@ -0,0 +1,132 @@
+import React from "react";
+
+import "styles/FeatureCalloutWidget.scss";
+import FocusTrap from "focus-trap-react";
+import { withI18n, withI18nProps } from "@lingui/react";
+import { getFeatureCalloutContent } from "./FeatureCalloutContent";
+import { useState } from "react";
+import { t } from "@lingui/macro";
+
+const FeatureCalloutWidget = withI18n()((props: withI18nProps) => {
+  const [isWidgetOpen, setWidgetVisibility] = useState(true);
+  const toggleWidget = () => setWidgetVisibility(!isWidgetOpen);
+
+  const [entryIndex, setEntryIndex] = useState(0);
+
+  const { i18n } = props;
+
+  const content = getFeatureCalloutContent();
+  const numberOfEntries = content.length;
+
+  let goToNextEntry = () => {
+    entryIndex === numberOfEntries - 1 ? setEntryIndex(0) : setEntryIndex(entryIndex + 1);
```

Comment:
It might be a bit less verbose if you put `setEntryIndex` on the outside of this, e.g.:

```js
setEntryIndex(entryIndex === numberOfEntries - 1 ? 0 : entryIndex + 1)
```

Although for that matter, an even more idiomatic way of doing this is to use the modulus operator, e.g.

```js
setEntryIndex((entryIndex + 1) % numberOfEntries)
```

You can apply similar logic to `goToPrevEntry` below.

Reply:
math math math math math